src/java/org/apache/cassandra/index/sai/IndexContext.java

```diff
@@ -354,6 +371,11 @@ public boolean isLiteral()
         return TypeUtil.isLiteral(getValidator());
     }
 
+    public boolean isVector()
+    {
+        return getValidator().isVector() && ((VectorType<?>)getValidator()).elementType instanceof FloatType;
```

Comment:
At the moment if an index is created on a vector column the type of the created index will depend on the type of  the vector elements. If the vector is made of floats we will create and ANN index that doesn't support equality restrictions. If the vector is made of any other thing we wil create a literal index that does support equalities.

I think this might be a bit confussing for users. They will be able to build indexes for equalities on any kind of vectors but float vectors. I guess we could:

1. Forbid SAI indexes in non-float vectors, so vectors are always indexed by ANN if ANN supports their element type (float)
2. Vectors are indexed as literals by default, unless some syntax or index property indicates that they should be indexed by ANN.

wdyt?

Reply:
I'm in favour of 1 for the time being. I don't think treating them as literals makes sense at all. If we were going to index them, using an index other than ANN, it's more likely that we would follow the collections path and effectively tokenize them, or something new.

I'll add some validation to block this case.

Reply:
Yes, that makes sense. I guess that if we were to ever allow indexing vectors as literal one might consider making it the default, so the most basic equality indexing is the default. But ANN seems much more useful in practice, so I guess it's better to have it as the default.